storage/backend_key.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"encoding/binary"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// backendKeyBytesLen is the length of backendKey bytes.
+// First a few bytes stores the revision of backendKey. The following byte
+// is a ' ' as delimiter. The last 4 bytes is the event type in big-endian
+// format.
+const backendKeyBytesLen = revBytesLen + 1 + 4
+
+type backendKey struct {
+	rev revision
+	t   storagepb.Event_EventType
```

Comment:
The reason to remove the concept from kv layer is to give the watchableKV the flexibility to define whatever event it wants without breaking the ondisk format at kv layer. Right now, they are the same structure and concept. When you want to change anything in the watchable layer, it breaks the kv layer. This is basically wrong and we need to change it.